Sources/Tokenizers/Tokenizer.swift

```diff
@@ -117,17 +124,15 @@ public protocol Tokenizer {
     var eosTokenId: Int? { get }
     var unknownToken: String? { get }
     var unknownTokenId: Int? { get }
-    
-    func applyChatTemplate(messages: [[String: String]]) throws -> [Int]
 
-    func applyChatTemplate(messages: [[String: String]], chatTemplate: String) throws -> [Int]
+    func applyChatTemplate(messages: [[String: String]]) throws -> [Int]
 
-    func applyChatTemplate(messages: [[String: String]], chatTemplateName: String) throws -> [Int]
+    func applyChatTemplate(messages: [[String: String]], chatTemplate: ChatTemplateArgument) throws -> [Int]
```

Comment:
I like this solution. But it's a breaking change, do you know if the current method is used by mlx or others?

Perhaps we could add an overload `applyChatTemplate(messages: [[String: String]], chatTemplateName: String)` that just builds a `ChatTemplateArgument` (`literal`) and forwards the call.

Reply:
I searched the mlx-swift and mlx-swift-examples repos and didn't find any instances of `applyChatTemplate` there. I have a [draft PR](https://github.dev/ml-explore/mlx-swift-examples) to use the chat templates once this PR and the Jinja PR land.

I think you mean the additional overload should have the argument `chatTemplate: String`. Sure, I can add that.